src/Console/Base/BaseEntityFullSyncCommand.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace SyncTools\Console\Base;
+
+use Illuminate\Console\Command;
+use SyncTools\Exceptions\ResourceGatewayConnectionException;
+use SyncTools\Gateways\ResourceGatewayInterface;
+use SyncTools\Repositories\CachedEntityRepositoryInterface;
+
+abstract class BaseEntityFullSyncCommand extends Command
+{
+    /**
+     * @throws ResourceGatewayConnectionException
+     */
+    public function handle(): void
+    {
+        $entityRepository = $this->getEntityRepository();
+        $entityRepository->cleanupLastSyncDateTime();
+
+        foreach ($this->getResourceGateway()->getResources() as $resource) {
+            $entityRepository->save($resource);
+        }
+
+        $entityRepository->deleteNotSynced();
+    }
```

Comment:
I think some kind of transactions are still needed, to ensure the integrity. One option would be to move transaction a level deeper into the loop so that transaction will still control successful saving of one row (sync object) and for rows that were not successful (transaction failed) there should be some retry mechanism to ensure that these rows would be synced as well eventually

Reply:
@thenouan I’m not sure I understand why the transaction should be started within the loop? Why not start the transaction before the loop and close it afterwards?